typespec-tests/src/main/java/com/cadl/server/contoso/sub/ContosoClient.java

```diff
@@ -2,7 +2,7 @@
 // Licensed under the MIT License.
 // Code generated by Microsoft (R) TypeSpec Code Generator.
 
-package com.cadl.server;
+package com.cadl.server.contoso.sub;
```

Comment:
Result of the test.

For `name=Contoso.Sub.ContosoClient` in below
```ts
  @client({
    name: "Contoso.Sub.ContosoClient",
    service: Cadl.ContosoServer,
  })
```

The Client/AsyncClient/Builder is moved to `contoso.sub` package.